www/js/survey/input-matcher.js

```diff
@@ -116,5 +116,37 @@ angular.module('emission.survey.inputmatcher', ['emission.plugin.logger'])
     Logger.log("Returning mostRecentEntry "+printUserInput(mostRecentEntry));
     return mostRecentEntry;
   }
+
+  // return array of matching trip additions
+  im.getTripAdditionsForTrip = function(trip, tripAdditionList) {
+    if (tripAdditionList === undefined) {
+      Logger.log("In getTripAdditionsForTrip, no trip addition input, returning []");
+      return undefined;
+    }
+
+    // remove deleted additions
+    tripAdditionList.forEach(addition => {
+        if (addition.data.status == "DELETED") {
+            // find and remove the original addition
+            const i = tripAdditionList.findIndex(a =>
+                a.data.end_ts == addition.data.end_ts
+                && a.data.start_ts == addition.data.start_ts);
+            tripAdditionList.splice(i, 1)
+            // also remove this deletion signifier
+            tripAdditionList.splice(tripAdditionList.indexOf(addition), 1);
+        }
+    });
```

Comment:
I remember.

Whichever approach is taken, doesn't the filtering need to also happen on the phone for unprocessed additions? Here I am using start_ts and end_ts

Reply:
Correct, the filtering does need to happen on the phone for unprocessed additions. I just think that the matching code on both the phone and the server needs to be consistent.

Reply:
- if we do use id, we need a separate id since unprocessed additions will not have the mongodb id

so let's think about pros and cons briefly:
- advantage of ids is that we can match even if the start and end ts are different (e.g. if we support edits in the future)
- advantage of `start_ts` and `end_ts` is that we can be consistent with the other matchings

So if we are going to support edits in which people can change their time range on edit, then we must use ids. This will also be very complicated, so I don't really want to do it.